Abstract Code from data loading to EDA

sapolraadnui opened the pull request:
Hi all,

Please kindly review this PR! Here are the key highlights of what I have done:

1. I added 3 py files which are code abstraction from .ipynb to .py files in the src folder.
- 01_data_loading.py
- 02_data_transformation.py
- 03_eda.py
I used 01, 02, and 03 to indicate the sequence that each code should be run
2. I left the functions defined in utils.py and validations.py as they were. No further adjustments were made as I only pulled the functions defined previously in the 3 new py files.
3. In the data folder, I created 2 folders: raw and processed to mimic Tiffany's repository to make it easier for the audience to understand.
4. I also create the results folder to store the figures created in EDA, models, and tables to tidy the repo and mimic Tiffany's repository.

These are all the updates from me. Let me know what you all think and if you want me to adjust anything. 

Thanks,
Paul

sasha-ss commented:
Hi Paul, 

Thank you for all your work! I have a few questions:

1. In the example, the validation step is included in the data preprocessing script. Should we combine these two scripts, or is it fine to keep them separate?
2. In addition, I noticed that for the EDA script, the input is processed_training_data, but in the data transformation script, the output is cleaned_train_data. Could we clarify whether these refer to the same dataset?

Thanks!

sapolraadnui commented:
Thanks all!

I have updated the codes as suggested. Here are main updates:
1. I added a few lines to check if the directories exist when saving outputs.
2. I deleted my EDA py file since Eli already took care of that.
3. I have added some lines to save the table output as a csv file of the data summary. For other tables, I believe they are in Sasha's part but we can take care of that to ensure consistency.
4. I separated two scripts since that's how they specified in the instruction.

Thanks,
Paul

Eligoze75 reviewed (approved):
👍🏽
